distributions/aws/aws-istio-authz-envoy-filter/base/envoy-filter-kubeflow-userid.yaml

```diff
@@ -0,0 +1,150 @@
+
+                for i=1, string.len(bitstring), 8 do
+                    local bitstring_chunk = string.sub(bitstring, i, i+7)
+
+                    if string.len(bitstring_chunk) < 8 then
+                        break
+                    end
+
+                    local output_string_char = string.char(tonumber(bitstring_chunk, 2))
+                    output_string = output_string .. output_string_char
+                end
+
+                return output_string
+            end
+
+            local function decode_base64(base64_string)
+                local base64_bitstring = base64_string_to_bitstring(base64_string)
+                return bitstring_to_string(base64_bitstring)
+            end
+
+            local function extract_jwt_payload(jwt)
+                local claim_start = string.find(jwt, "%.")
+                if claim_start == nil then
+                    request_handling_error("Could not find claims section of JWT")
+                end
+
+                local claim_end = string.find(jwt, "%.", claim_start+1)
+                if claim_end == nil then
+                    request_handling_error("Could not find claims section of JWT")
+                end
+
+                -- don't include periods
+                return string.sub(jwt, claim_start+1, claim_end-1)
+            end
+
+            local function get_email_from_jwt_payload(jwt_payload)
+                local claim_start = 0
+                while true do
+                    local claim_end = string.find(jwt_payload, ",", claim_start)
+                    local claim = string.sub(jwt_payload, claim_start, claim_end)
+
+                    if string.find(claim, "\"email\"") ~= nil then
+                        local delimiter_index = string.find(claim, ":")
+                        if delimiter_index == nil then
+                            request_handling_error("Could not parse email in JWT payload")
+                        end
+                        local email_start = string.find(claim, "\"", delimiter_index+1)
+                        if email_start == nil then
+                            request_handling_error("Could not parse email in JWT payload")
+                        end
+                        local email_end = string.find(claim, "\"", email_start+1)
+                        if email_end == nil then
+                            request_handling_error("Could not parse email in JWT payload")
+                        end
+
+                        return string.sub(claim, email_start+1, email_end-1)
+                    end
+
+                    if claim_end == nil then
+                        break
+                    end
+
+                    claim_start = claim_end + 1
+                end
+    
+                request_handling_error("Could not find email in JWT payload")
+            end
+
+            function envoy_on_request(request_handle)
+              local jwt_head = request_handle:headers():get("x-amzn-oidc-data")
```

Comment:
Should we first check if `request_handle's header` has `kubeflow-userid` populated?